### PR TITLE
Update AdvancedSearch ReadMe

### PR DESCRIPTION
**Pull request by diazwatson**
### Description (*)

Updates ReadMe for the Magento_AdvancedSearch module.

### Fixed Issues (if relevant)

1. magento/devdocs https://github.com/magento/devdocs/issues/4451: README content for modules - work in M2 Code

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-5658](https://jira.corp.magento.com/browse/ENGCOM-5658) has been created to process this Pull Request

File: app/code/Magento/AdvancedSearch/README.md [hunk 1]
```diff
@@ -1 +1,39 @@
-AdvancedSearch module introduces advanced search functionality and provides interfaces that allow to implement this functionality by 3rd party search engines
+# Magento_AdvancedSearch module
+Magento_AdvancedSearch module introduces advanced search functionality and provides interfaces that allow to implement 
+this functionality by 3rd party search engines.
```
Comment (jeff-matthews) [inline app/code/Magento/AdvancedSearch/README.md]: Suggested rewrite:

> The Magento_AdvancedSearch module introduces advanced search functionality and provides interfaces that allow third-party search engines to implement this functionality.

File: app/code/Magento/AdvancedSearch/README.md [hunk 2]
```diff
@@ -1 +1,39 @@
...
+
+### Layouts
+
+The module interact with the following layout handles in the `view/adminhtml/layout` directory:
+
+- `catalog_search_block`
+- `catalog_search_edit`
+- `catalog_search_relatedgrid`
+
+The module interact with the following layout handles in the `view/frontend/layout` directory:
```
Comment (jeff-matthews) [inline app/code/Magento/AdvancedSearch/README.md]:
```suggestion
The module interacts with the following layout handles in the `view/frontend/layout` directory:
```

File: app/code/Magento/AdvancedSearch/README.md [hunk 3]
```diff
@@ -1 +1,39 @@
...
+
+This module observes the following event:
+
+ - `catalogsearch_query_save_after` in the `Magento\AdvancedSearch\Model\Recommendations\SaveSearchQueryRelationsObserver` file. 
+
+For information about an event in Magento 2, see [Events and observers](http://devdocs.magento.com/guides/v2.3/extension-dev-guide/events-and-observers.html#events).
+
+### Layouts
+
+The module interact with the following layout handles in the `view/adminhtml/layout` directory:
```
Comment (jeff-matthews) [inline app/code/Magento/AdvancedSearch/README.md]:
```suggestion
The module interacts with the following layout handles in the `view/adminhtml/layout` directory:
```

**Review by jeff-matthews — CHANGES_REQUESTED** [review]
Thanks for working on these READMEs @diazwatson! I made some minor editorial suggestions.

**Comment by diazwatson** [comment]
Thanks to you @jeff-matthews for taking the time to CR this.
Requested changes are now done.

**Review by jeff-matthews — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @jeff-matthews, thank you for the review.
[ENGCOM-5658](https://jira.corp.magento.com/browse/ENGCOM-5658) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
QA not applicable

**Pull request merged**